internal/support/logging/init.go

```diff
@@ -70,10 +71,21 @@ func (s ServiceInit) BootstrapHandler(
 	// update global variables.
 	LoggingClient = container.LoggingClientFrom(dic.Get)
 
+	// get database credentials.
+	var credentials config.Credentials
+	for startupTimer.HasNotElapsed() {
+		var err error
+		credentials, err = container.CredentialsProviderFrom(dic.Get).GetDatabaseCredentials(Configuration.Databases["Primary"])
+		if err == nil {
+			break
+		}
+		startupTimer.SleepForInterval()
```

Comment:
You should log the error when err != nil -- similar to what's being done below when getPersistence() returns an err.